web/Services/AdminService.cs

```diff
@@ -98,46 +98,50 @@ private async Task<List<UserDbModel>> GetPendingAdmins()
         return pendingAdmins;
     }
 
-    public async Task<List<LandlordDbModel>> GetLandlordDisplayList(string approvalStatus)
+    public async Task<List<LandlordDbModel>> GetLandlordList(LandlordListModel landlordListModel)
     {
-        return approvalStatus switch
+        var landlordQuery = _dbContext.Landlords.AsQueryable();
+        if (landlordListModel.IsApproved != null)
         {
-            "Unapproved" => await _dbContext.Landlords.Where(u => u.CharterApproved == false).ToListAsync(),
-            "Approved" => await _dbContext.Landlords.Where(u => u.CharterApproved == true).ToListAsync(),
-            _ => await _dbContext.Landlords.ToListAsync(),
-        };
-    }
-    
-    public async Task<List<TenantDbModel>> GetTenantList()
-    {
-        return await _dbContext.Tenants.ToListAsync();
+            landlordQuery = landlordQuery.Where(l => l.CharterApproved == landlordListModel.IsApproved);
+        }
+        if (landlordListModel.IsAssigned != null)
+        {
+            bool? target = landlordListModel.IsAssigned;
+            landlordQuery = landlordQuery.Where(l => _dbContext.Users.SingleOrDefault(u => u.LandlordId == l.Id) != null == target);

```

Comment:
```suggestion
            landlordQuery = landlordQuery.Where(l => _dbContext.Users.Any(u => u.LandlordId == l.Id) == landlordListModel.IsAssigned);

```

Reply:
Ok that does work - I think I was misunderstanding how Any worked, so was trying to use it in the wrong way.